memcached.c

```diff
@@ -3633,6 +3633,31 @@ static void process_stat(conn *c, token_t *tokens, const size_t ntokens) {
     }
 }
 
+static void process_close_connection(conn *c, token_t *tokens,
+    const size_t ntokens) {
+    set_noreply_maybe(c, tokens, ntokens);
+    uint32_t connid;
+
+    if (!safe_strtoul(tokens[1].value, &connid) || connid >= max_fds) {
+        out_string(c, "ERROR Invalid connection id");
+        return;
+    }
+
+    conn* conn = conns[connid];
+    if (conn == NULL) {
+        out_string(c, "ERROR Connection not found");
+    } else if (conn->state == conn_listening) {
+        out_string(c, "ERROR Listening connection");
+    } else if (IS_UDP(conn->transport) && conn->state == conn_read) {
+        out_string(c, "ERROR UDP connection");
+    } else if (conn->sfd == c->sfd) {
+        out_string(c, "ERROR Current connection");
+    } else {
+        conn_close(conn);
```

Comment:
Should we close the connections asynchronously?

Reply:
haven't looked closely yet; if you do it the same way the idle connection kicker works it should be safe. IIRC that was a signal back to the owner thread to close it out?

Reply:
Yes it is. Let me do it that way.